src/clockcache.c

```diff
@@ -1816,20 +1816,25 @@ clockcache_init(clockcache        *cc,   // OUT
    cc->heap_id = hid;
 
    /* lookup maps addrs to entries, entry contains the entries themselves */
-   cc->lookup =
-      TYPED_ARRAY_MALLOC(cc->heap_id, cc->lookup, allocator_page_capacity);
+   platform_memfrag memfrag_cc_lookup;
+   cc->lookup = TYPED_ARRAY_MALLOC_MF(
+      cc->heap_id, cc->lookup, allocator_page_capacity, &memfrag_cc_lookup);
    if (!cc->lookup) {
       goto alloc_error;
    }
+   cc->lookup_size = memfrag_size(&memfrag_cc_lookup);
+
    for (i = 0; i < allocator_page_capacity; i++) {
       cc->lookup[i] = CC_UNMAPPED_ENTRY;
    }
 
-   cc->entry =
-      TYPED_ARRAY_ZALLOC(cc->heap_id, cc->entry, cc->cfg->page_capacity);
+   platform_memfrag memfrag_cc_entry;
+   cc->entry = TYPED_ARRAY_ZALLOC_MF(
+      cc->heap_id, cc->entry, cc->cfg->page_capacity, &memfrag_cc_entry);
    if (!cc->entry) {
       goto alloc_error;
    }
+   cc->entry_size = memfrag_size(&memfrag_cc_entry);
```

Comment:
Alternative: One thing I considered was to require that all such structures have a commonly-named `mf_size` field and this caller-macro will (behind-the-scenes) stash away `memfrag->size` into this structure's`mf_size` field.

But that solution hides some of this machinations behind macros, which might make code tracking difficult.

Also, not all structures need, or even have, this `mf_size` field. You only need this tracking `mf_size` if the free is done in some entirely different function. 

For most usages, where allocation and free is done in the same function, we just directly use `memfrag->size` to do the free. In such cases, the allocating structure (or memory ptr) will not need, and will not have, a `mf_size` field.

Hence, this alternative hasn't been implemented.